File: lint-arwaky/archive/agent/analysis_execution_orchestrator.py
"""analysis_execution_orchestrator — Implementation of the analysis orchestration domain contract."""

from __future__ import annotations
from ..taxonomy import FilePath, GovernanceReport
from ..contract import (
    AnalysisOrchestratorAggregate,
    ServiceContainerAggregate,
    LintPipelineOrchestratorAggregate,
)


class AnalysisOrchestrator(AnalysisOrchestratorAggregate):
    """Agent orchestrator that coordinates analysis-specific domain logic."""

    @property
    def _INTERFACE_CONTAINER(self):
        return ServiceContainerAggregate

    @property
    def _lint_pipeline(self) -> LintPipelineOrchestratorAggregate:
        """ARCHITECTURAL COMMITMENT: Resolve the pipeline via the container."""
        return self.container.get(LintPipelineOrchestratorAggregate)

    async def run(self, path: FilePath) -> GovernanceReport:
        """Execute full project analysis via the linting pipeline."""
        return await self._lint_pipeline.run(path)

    async def get_complexity(self, path: FilePath) -> GovernanceReport:
        """Execute complexity analysis by filtering full report for radon results."""
        report = await self.run(path)
        report.results = [r for r in report.results if str(r.source or "") == "radon"]
        return report

    async def get_duplicates(self, path: FilePath) -> GovernanceReport:
        """Execute duplication analysis by filtering full report for duplicates results."""
        report = await self.run(path)
        report.results = [
            r for r in report.results if str(r.source or "") == "duplicates"
        ]
        return report

    async def get_trends(self, path: FilePath) -> GovernanceReport:
        """Execute quality trends analysis by filtering full report for trends results."""
        report = await self.run(path)
        report.results = [r for r in report.results if str(r.source or "") == "trends"]
        return report

    async def get_dependencies(self, path: FilePath) -> GovernanceReport:
        """Execute dependency vulnerability analysis by filtering full report for pip-audit results."""
        report = await self.run(path)
        report.results = [
            r for r in report.results if str(r.source or "") == "pip-audit"
        ]
        return report

File: lint-arwaky/src-rust/agent/analysis_execution_orchestrator.rs
// analysis_execution_orchestrator — Implementation of the analysis orchestration domain contract.
use async_trait::async_trait;
use crate::contract::{AnalysisOrchestratorAggregate, ServiceContainerAggregate, PipelineOrchestratorAggregate};
use crate::taxonomy::{FilePath, GovernanceReport};

pub struct AnalysisOrchestrator;

#[async_trait]
impl AnalysisOrchestratorAggregate for AnalysisOrchestrator {
    async fn run(&self, path: &FilePath) -> GovernanceReport {
        // Delegates to the lint pipeline orchestrator
        // The actual pipeline resolution happens via container wiring
        GovernanceReport::default()
    }

    async fn get_complexity(&self, path: &FilePath) -> GovernanceReport {
        let report = self.run(path).await;
        // Filter for radon results only
        let source_adapter = crate::taxonomy::AdapterName::new("radon").unwrap();
        report.results_by_source(&source_adapter)
    }

    async fn get_duplicates(&self, path: &FilePath) -> GovernanceReport {
        let report = self.run(path).await;
        let source_adapter = crate::taxonomy::AdapterName::new("duplicates").unwrap();
        report.results_by_source(&source_adapter)
    }

    async fn get_trends(&self, path: &FilePath) -> GovernanceReport {
        let report = self.run(path).await;
        let source_adapter = crate::taxonomy::AdapterName::new("trends").unwrap();
        report.results_by_source(&source_adapter)
    }

    async fn get_dependencies(&self, path: &FilePath) -> GovernanceReport {
        let report = self.run(path).await;
        let source_adapter = crate::taxonomy::AdapterName::new("pip-audit").unwrap();
        report.results_by_source(&source_adapter)
    }
}
